src/azul/files.py

```diff
@@ -15,3 +16,8 @@ def write_file_atomically(path, mode=0o644):
     except BaseException:
         os.unlink(temp_path)
         raise
+
+
+def file_sha(path):
```

Comment:
```suggestion
def file_sha1(path):
```

Reply:
The way this is written, its not general enough to deserve a place in this module. Either rewrite it to support streaming and therefore large files which should be relatively easy (use a chunk size 1M) or inline it. AFAICT, there is only one call site. If you go the former route, consider adding doctests, one for `"/dev/null"` and one for `__file__`. I'd also accept a docstring that warns about the caveat of memory consumption for large files but consider it lame since streaming should be easy to add.

Reply:
I considered adding a doctest for `__file__` but decided that finding a collision for sha1 was a bit out of scope for this PR.